Fix translated host, allow ipv6

fthyssen opened the pull request:
These changes address the use of different sockets, and an issue with translated hostnames.
Translated hostnames and `Counter64`/`Uinteger32` to `int` were needed, and thus tested and used.

The other (ipv6, unix socket) less so, so feel free to ask me to drop those changes if they are out of scope.

hh-h commented:
Hey! Thank you for the changes. Could you explain the issue with translated hostnames?  
Also, all changes should be covered with tests, so I need to think about how to reorganize the test flow so it will be possible to test things such as sockets...

fthyssen commented:
The issue was that the key used in `datagram_received` didn't match the one in `self.requests`, so it timed out waiting for a reply.

I added it to the existing tests, and removed the unix socket - I realized after testing that connecting to one wasn't supported.  I did leave `SocketKey`, just in case you extend capabilities at some point.

fthyssen commented:
It looks like ipv6 is not supported by the test environment.  Tests did pass with asyncio on my machine.
Is this something you'd like to fix in the docker image, or from the test scripts?

hh-h commented:
> Is this something you'd like to fix in the docker image, or from the test scripts?

I'm not sure, the current docker way is pretty ugly because I have to use very old version of `tox-docker==1.2.1` library. I'll check what can be done with `tox-docker==2.0.0`.

> The issue was that the key used in datagram_received didn't match the one in self.requests, so it timed out waiting for a reply.

ah, like `localhost` or `example.com`?

fthyssen commented:
Correct, like `localhost` and `example.com`

If you can install and run netsnmp, you could replace the docker-instance as follows:
- extract `/etc/snmp/passtest` from the docker image and add it to this repo
- add `snmpd.conf` to this repo as

```
agentAddress udp:8161
agentAddress udp6:8161

rocommunity public
rwcommunity private
rocommunity6 public
rwcommunity6 private

pass .1.3.6.1.4.1.8072.2.255 /bin/sh passtest
```

- add `snmpd` as an apt-addon-package to travis.yml
- run `snmpd -f -c snmpd.conf &` before running tests

Also, adjust the test-port accordingly.

This would make the tests a bit more transparent and easy to extend as well.

hh-h commented:
> If you can install and run netsnmp, you could replace the docker-instance as follows:

Unfortunately, this will break local tox.

Also, seems like Travis doesn't support ipv6... https://docs.travis-ci.com/user/reference/overview/#virtualisation-environment-vs-operating-system

hh-h commented on aiosnmp/snmp.py:
```diff
@@ -51,7 +51,8 @@ async def _send(self, message: SnmpMessage) -> List[SnmpVarbind]:
             await self._connect()
         assert self._protocol
-        return await self._protocol._send(message, self.host, self.port)
+        assert self._peername is not None
```
just `assert self._peername`

hh-h commented on aiosnmp/protocols.py:
```diff
@@ -45,6 +45,9 @@
 }
 
+Address = Union[Tuple[str, int], Tuple[str, int, int, int]]
+SocketKey = Tuple[str, int]
```
let's get rid of `SocketKey`

hh-h commented:
I migrated to azure pipelines, seems like it should work now. Please rebase with master and squash everything to one commit.

hh-h commented:
Thank you! Let me figure out how to deploy with azure pipelines, then the new build will be on pypi.
